server/src/main/java/de/uftos/repositories/ucdl/parser/UcdlParser.java

```diff
@@ -0,0 +1,119 @@
+
+    Iterator<Map.Entry<String, JsonNode>> iterator = jsonNode.fields();
+    while (iterator.hasNext()) {
+      Map.Entry<String, JsonNode> entry = iterator.next();
+      try {
+        constraints.put(entry.getKey(),
+            parseConstraintDefinition(entry.getKey(), entry.getValue()));
+      } catch (ParseException e) {
+        throw new ParseException(
+            "Error in constraint definition \"" + entry.getKey() + "\":" + e.getMessage());
+      }
+    }
+    return constraints;
+  }
+
+  //parses one full definition
+  private static ConstraintDefinitionDto parseConstraintDefinition(String name,
+                                                                   JsonNode constraintDefinition)
+      throws ParseException {
+    if (constraintDefinition.get("description") == null) {
+      throw new ParseException("Constraint description of constraint " + name + " is missing!");
+    }
+    if (constraintDefinition.get("default_type") == null) {
+      throw new ParseException("Default type of constraint " + name + " is missing!");
+    }
+    if (constraintDefinition.get("parameter") == null) {
+      throw new ParseException("Parameters of constraint " + name + " are missing!");
+    }
+    if (constraintDefinition.get("definition") == null) {
+      throw new ParseException("Definition of constraint " + name + " is missing!");
+    }
+
+    String description = constraintDefinition.get("description").textValue();
+
+    RewardPenalize defaultType = switch (constraintDefinition.get("default_type").textValue()) {
+      case "HARD_REWARD" -> RewardPenalize.HARD_REWARD;
+      case "SOFT_REWARD" -> RewardPenalize.SOFT_REWARD;
+      case "HARD_PENALIZE" -> RewardPenalize.HARD_PENALIZE;
+      case "SOFT_PENALIZE" -> RewardPenalize.SOFT_PENALIZE;
+      case null -> RewardPenalize.SOFT_PENALIZE;
+      default -> throw new ParseException(
+          "Invalid default_type. \"HARD_REWARD\", \"SOFT_REWARD\", \"HARD_PENALIZE\","
+              + " \"SOFT_PENALIZE\", or empty field expected!");
+    };
+
+    HashMap<String, ResourceType> parameters = new HashMap<>();
+    constraintDefinition.get("parameter").fields().forEachRemaining((entry) -> {
+      if (entry != null) {
+        try {
+          parameters.put(entry.getKey(), getResourceType(entry.getValue().textValue()));
+        } catch (ParseException e) {
+          throw new RuntimeException(e);
+        }
+      }
+    });
+    AbstractSyntaxTreeDto definition =
+        parseDefinition(constraintDefinition.get("definition").textValue(), parameters);
+
+    return new ConstraintDefinitionDto(name, description, defaultType, parameters, definition);
+  }
+
+  private static AbstractSyntaxTreeDto parseDefinition(String definition,
+                                                       HashMap<String, ResourceType> parameters)
+      throws de.uftos.repositories.ucdl.parser.javacc.ParseException {
+    return DefinitionParser.parseDefinition(definition, parameters);
+  }
```

Comment:
Why is this method needed? AFAICT it's only called once and literally only has one line of code, this doesn't even help with readability